lib/spack/spack/solver/asp.py

```diff
@@ -1290,6 +1271,9 @@ class Body(object):
 
                 clauses.append(f.variant_value(spec.name, vname, value))
 
+                if variant.propagate:
+                    clauses.append(f.variant_propagate(spec.name, vname))
```

Comment:
Eventually we should make the propagation be per-flag, but that's future work in another PR. Then we can enable

```
spack install foo cflags=-g cflags==-O3
```
to have the root build with `-g -O3` and the dependencies just with `-O3`.

Reply:
So do you think `node_flag_propagate` should include the flag_value in the future?

Reply:
Yes, exactly